### ISSUE TITLE
Peer dependency issue - @cypress/webpack-preprocessor doesn't support webpack@^5

### ISSUE THREAD

**Issue by Laurent147** [cypress-io/cypress#16401]
## Current behavior
Installation of @cypress/react fails due to unresolved peer dependency. 

### Pre-existing install of webpack 5 & react 17
From a repository with the following packages:
```json
  "dependencies": {
    "html-webpack-plugin": "^5.3.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "webpack": "^5.36.2",
    "webpack-cli": "^4.7.0",
    "webpack-dev-server": "^3.11.2"
  }
```
 when running the [cypress guide command](https://docs.cypress.io/guides/component-testing/introduction#Install) (minus webpack-dev-server):
```bash
npm install --save-dev cypress @cypress/react @cypress/webpack-dev-server
```
install fails with error: "Enable to resolve dependency tree" (see below screenshot)
The culprit seems to be @cypress/webpack-preprocessor which can't support a webpack version greater than 4.

<img src="https://i.ibb.co/cw0ZYb2/Screen-Shot-2021-05-07-at-11-51-01-AM.png" alt="Screen-Shot-2021-05-07-at-11-51-01-AM" border="0">

### Pre-existing install of react 17 only
From an empty repository, the install fails due to html-webpack-plugin ^5.3.1 needing webpack 5 rather than webpack 4

**command**
```bash
npm install --save-dev cypress @cypress/react @cypress/webpack-dev-server webpack-dev-server
```
**Logs**
<img src="https://i.ibb.co/PGnprcW/Screen-Shot-2021-05-07-at-1-49-44-PM.png" alt="Screen-Shot-2021-05-07-at-1-49-44-PM" border="0">

### Pre-existing install of react 17 & html-webpack-plugin 4
From the only the following repository, the install works:
```json
  "dependencies": {
    "html-webpack-plugin": "^4.5.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
  }
```

**command**
```bash
npm install --save-dev cypress @cypress/react @cypress/webpack-dev-server webpack-dev-server
```

**Logs**
<img src="https://i.ibb.co/kxBk5rJ/Screen-Shot-2021-05-07-at-1-57-52-PM.png" alt="Screen-Shot-2021-05-07-at-1-57-52-PM" border="0">

## Environment
* node: v14.16.0
* npm: v7.11.2
* Mac OS Big Sur version 11.3

## Expected behavior
Clean install with support for webpack 5

**Comment by lmiller1990** [comment]
Hm, I don't think that `@cypress/webpack-preprocessor` should even be a dependency of `@cypress/react` anymore: https://github.com/cypress-io/cypress/blob/1e7a3faa4c26bea5aea12be4f5eb023235c3b1a4/npm/react/package.json#L21

I will investigate this.

**Comment by lmiller1990** [comment]
I was not able to reproduce this. I ran:

```sh
echo '{ 
 "dependencies": { 
    "html-webpack-plugin": "^5.3.1", 
    "react": "^17.0.2", 
    "react-dom": "^17.0.2", 
    "webpack": "^5.36.2", 
    "webpack-cli": "^4.7.0", 
    "webpack-dev-server": "^3.11.2" 
  } 
}' > package.json && npm install --save-dev cypress @cypress/react @cypress/webpack-dev-server
```

Everything installs ok.

```
$ node -v
v14.16.0
$ npm -v
6.14.11
$
```

Maybe I need npm v7.

**Comment by lmiller1990** [comment]
This should help: https://github.com/cypress-io/cypress/pull/16412

**Comment by lmiller1990** [comment]
Fixed in #16412, `@cypress/react":"^5.6.0"`. It's been released. I re-ran the above command and it worked.

**Comment by Laurent147** [comment]
@lmiller1990 thanks for the very reactive fix. I tested the install on my side and all seems to work with the exact command suggested by your guide.
Thanks!
